crates/ruff_python_formatter/src/comments/visitor.rs

```diff
@@ -439,6 +439,14 @@ impl<'a> DecoratedComment<'a> {
     pub(super) fn text_position(&self) -> CommentTextPosition {
         self.text_position
     }
+
+    /// Change the position
+    pub(super) fn change_position(self, text_position: CommentTextPosition) -> Self {
```

Comment:
Nit: I think taking a `&mut self` would be fine here. It could simplify some of your call sites .